client/gutenberg/editor/edit-post/components/header/header-toolbar/index.js

```diff
@@ -23,30 +29,110 @@ import {
 	NavigableToolbar,
 } from '@wordpress/editor';
 
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Drafts from 'layout/masterbar/drafts';
+import Site from 'blocks/site';
+import { addSiteFragment } from 'lib/route';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getSelectedSite } from 'state/ui/selectors';
+import { getRouteHistory } from 'state/ui/action-log/selectors';
+
+function getCloseButtonPath( routeHistory, site, post ) {
+	// @see post-editor/editor-ground-control/index.jsx
+	const lastNonEditorPath = findLast(
+		routeHistory,
+		action => ! action.path.match( /^\/gutenberg\/(post|page|(edit\/[^\/]+))\/[^\/]+(\/\d+)?$/i )
+	);
+	if ( lastNonEditorPath ) {
+		return lastNonEditorPath.path;
+	}
+
+	// @see post-editor/post-editor.jsx
+	const { type } = post;
+	let path = '';
+	switch ( type ) {
+		case 'page':
+			path = '/pages';
+			break;
+		case 'post':
+			path = '/posts';
+			break;
+		default:
+			path = `/types/${ type }`;
+	}
+	if ( type === 'post' && site && ! site.jetpack && ! site.single_user_site ) {
```

Comment:
Really minor nit, but consider only fetching specific site attributes in mapStateToProps. This can help us audit what things we need from our bloated site object.

Reply:
The `site` object is also used by `<Site>`, so we need to have it around anyway.